### DISCUSSION TITLE
Where to use ReactPHP and Framework-X

### DISCUSSION THREAD

**Discussion by carliedu** [clue/framework-x#229] [Ideas]
What type of system should I use ReactPHP and Framework-X for?
This question I'm asking myself for many months.
I discovered ReactPHP when searching on Google, how to implement a little system to capture Bitcoin quotes (the first idea was very specific), do some calculations and decide whether to buy or sell. Very simple and objective.
But, little by little, I realized that it was much more complex than that.
First, with traditional (synchronous) programming that I used, this would not work (it would, but the action/reaction would take so long, that the quote would already have changed and the entire calculation would no longer have valid results, one millisecond after the routine ended.
That's when, researching options, I ended up discovering ReactPHP.
But in reality, I found only the tiny tip of a monstrous iceberg.
I took over 6 months until I understood things an got a very embryonic system, that get data with Ratchet, treat them and display it at an browser interface real-time.
My idea of system is that it has to have an real time user interfarce without the need of submition needs. And, for now I only can think of Trading.
So I want to launch here a discussion with suggestions, case reports of applications already implemented and in project with ReactPHP and Framework-X.

Best, @carliedu

**Comment by SimonFrings** [comment]
Hey @carliedu, good questions as I'm also interested in what type of applications are built with Framework X and ReactPHP :+1:

There are a lot of different use cases out there for both of these projects, especially when it comes to clients handling hundreds or thousands of concurrent connections, long-running applications, fast database access, and much more. I agree that trading is one of the recently seen use cases as reacting to real-time events is very important there and thus presents a great use case to show ReactPHP's speed and efficiency.

We ourselves use Framework X and ReactPHP in many of our projects, especially in cases where we consume large amounts of data. We haven't communicated about many of these projects to the outside world as we created them to improve our own workflows and haven't yet decided what will happen with them (whether to release them as open source or use them as commercial projects).

In order to get a better overview of projects using Framework X and ReactPHP, I created the "Show and Tell" section when activating discussion for this repository. This hasn't seen much activity yet (maybe because people don't know it exists), but I hope this will gain some more traction over time.

We also have a list of third-party projects that use ReactPHP. Definitely worth taking a look at them as well: https://github.com/reactphp/reactphp/wiki/Users

Hope this gives some more insights :+1:
